hedera-node/src/main/java/com/hedera/services/context/BasicTransactionContext.java

```diff
@@ -332,6 +335,16 @@ public void setCreateResult(final EvmFnResult result) {
 		recordConfig = expiringRecord -> expiringRecord.setContractCreateResult(result);
 	}
 
+	@Override
+	public void setSidecarRecords(final List<TransactionSidecarRecord.Builder> sidecars) {
```

Comment:
Would prefer to make this,
```
public void addSidecarRecords(final TransactionSidecarRecord.Builder sidecar)
```
as then we can make `sidecarRecords` a `final` and avoid creating (at least) two extra `ArrayList` objects per contract operation.

Reply:
Makes sense, fixed with [this commit](https://github.com/hashgraph/hedera-services/pull/3625/commits/8f9b60282b5d4090d2c86749bf5597b5e33e2fc2).